solarforecastarbiter/reports/figures.py

```diff
@@ -0,0 +1,211 @@
+        timestamp, observation, forecast.
+    start : pandas.Timestamp
+        Report start time
+    end : pandas.Timestamp
+        Report end time
+
+    Returns
+    -------
+    fig : bokeh.plotting.figure
+    """
+
+    palette = iter(PALETTE)
+
+    fig = figure(
+        sizing_mode='scale_width', plot_width=900, plot_height=300,
+        x_range=(start, end), x_axis_type='datetime',
+        tools='pan,wheel_zoom,reset',
+        name='timeseries')
+
+    plotted_objects = []
+    for fx_obs, cds in fx_obs_cds:
+        if fx_obs.observation in plotted_objects:
+            pass
+        else:
+            plotted_objects.append(fx_obs.observation)
+            plot_method, kwargs = line_or_step(
+                fx_obs.observation.interval_label)
+            name = _obs_name(fx_obs)
+            getattr(fig, plot_method)(
+                x='timestamp', y='observation', source=cds,
+                color='black', legend=name,  **kwargs)
+        if fx_obs.forecast in plotted_objects:
+            pass
+        else:
+            plotted_objects.append(fx_obs.forecast)
+            plot_method, kwargs = line_or_step(
+                fx_obs.forecast.interval_label)
+            name = _fx_name(fx_obs)
+            getattr(fig, plot_method)(
+                x='timestamp', y='forecast', source=cds,
+                color=next(palette), legend=name,  **kwargs)
+
+    fig.legend.location = "top_left"
+    fig.legend.click_policy = "hide"
+    fig.xaxis.axis_label = 'Time (UTC)'
+    fig.yaxis.axis_label = format_variable_name(fx_obs.forecast.variable,
+                                                fx_obs.forecast.units)
+    return fig
+
+
+def _get_scatter_limits(fx_obs_cds):
+    extremes = []
+    for _, cds in fx_obs_cds:
+        for kind in ('forecast', 'observation'):
+            extremes.append(cds.data[kind].min())
+            extremes.append(cds.data[kind].max())
+    return min(extremes), max(extremes)
+
+
+def scatter(fx_obs_cds):
+    """
+    Scatter plot of one or more forecasts and observations.
+
+    Parameters
+    ----------
+    obs_fx_cds : tuple of (ForecastObservation, cds) tuples
+        ForecastObservation is a datamodel.ForecastObservation object.
+        cds is a Bokeh ColumnDataSource with columns
+        timestamp, observation, forecast.
+
+    Returns
+    -------
+    fig : bokeh.plotting.figure
+    """
+    xy_min, xy_max = _get_scatter_limits(fx_obs_cds)
+
+    fig = figure(
+        plot_width=450, plot_height=400, match_aspect=True,  # does not work?
+        x_range=Range1d(xy_min, xy_max), y_range=Range1d(xy_min, xy_max),
+        tools='pan,wheel_zoom,reset',
+        name='scatter')
+
+    kwargs = dict(size=6, line_color=None)
+
+    palette = iter(PALETTE)
+
+    for fx_obs, cds in fx_obs_cds:
+        fig.scatter(
+            x='observation', y='forecast', source=cds,
+            fill_color=next(palette), legend=fx_obs.forecast.name, **kwargs)
```

Comment:
again if only using a single observation: maybe the best approach is to combine the observation and all forecasts into a single CDS, also avoids duplicating timestamp

Reply:
I didn't want to assume that all forecasts have the same interval length.

Reply:
hmm, that's a good point.